lib/spack/spack/mirror.py

```diff
@@ -68,21 +58,72 @@ def mirror_archive_filename(spec, fetcher, resource_id=None):
         # Otherwise we'll make a .tar.gz ourselves
         ext = 'tar.gz'
 
-    if resource_id:
-        filename = "%s-%s" % (resource_id, spec.version) + ".%s" % ext
-    else:
-        filename = "%s-%s" % (spec.package.name, spec.version) + ".%s" % ext
+    return ext
+
+
+class MirrorReference(object):
```

Comment:
needs a docstring indicating this stores all the places in a mirror where a particular file could be.